services/search/src/search/routes/filter.py

```diff
@@ -188,7 +188,7 @@ def execute_filter_query(
 ) -> tuple[int, pa.Table]:
     with duckdb_connect(database=index_file_location) as con:
         filter_query = FILTER_QUERY.format(
-            columns="*" if columns == "*" else ",".join([f'"{column}"' for column in [ROW_IDX_COLUMN] + columns]),
+            columns="*" if columns == "*" else ",".join([f'"{column}"' for column in columns]),
```

Comment:
Thanks for this fix.

I guess this bug was introduced by this PR:
- #2347
- See my comment: https://github.com/huggingface/datasets-server/pull/2347#discussion_r1469701184

IMHO, I find misleading to use the word "features" with 2 different meanings:
- the meaning I was supporting in my comment to that PR: "features" refers to the _original_ dataset features, those handled exclusively by the `datasets` library
- now we use "features" with a different meaning: all the columns stored in the DuckDB index file, which contain the original dataset features, plus any additional auxiliary columns required by datasets-server (like, "__hf_index_id", and any other additional column we may need in the future)

CC: @lhoestq for visibility